js/iframe-coordinator/src/messages/ClientToHost.ts

```diff
@@ -1,57 +1,18 @@
-import {
-  constant,
-  either3,
-  guard,
-  mixed,
-  object,
-  optional,
-  string
-} from 'decoders';
-import { LabeledNavRequest } from './NavRequest';
-import { LabeledPublication } from './Publication';
-import { LabeledToast } from './Toast';
+import { LabeledNavRequest, validateNavRequest } from './NavRequest';
+import { LabeledPublication, validatePublication } from './Publication';
+import { LabeledToast, validateToast } from './Toast';
 
 export type ClientToHost =
   | LabeledPublication
   | LabeledToast
   | LabeledNavRequest;
 
-const labeledPublicationDecoder = object({
-  msgType: constant('publish'),
-  msg: object({
-    topic: string,
-    payload: mixed
-  })
-});
-
-const labeledToastDecoder = object({
-  msgType: constant('toastRequest'),
-  msg: object({
-    title: optional(string),
-    message: string,
-    custom: mixed
-  })
-});
-
-const labeledNavRequestDecoder = object({
-  msgType: constant('navRequest'),
-  msg: object({
-    url: string
-  })
-});
-
-const decoder = guard(
-  either3(
-    labeledPublicationDecoder,
-    labeledToastDecoder,
-    labeledNavRequestDecoder
-  )
-);
-
 export function validate(msg: any): ClientToHost | null {
-  try {
-    return decoder(msg);
-  } catch {
+  if (!msg || !msg.msgType || !msg.msg) {
     return null;
   }
+
+  return (
+    validateNavRequest(msg) || validatePublication(msg) || validateToast(msg)
```

Comment:
This is a little hard to read if you don't know what the validate methods do.  I considered adding another util for like validateOneOf(vNR, vP, vT) but I don't think that helped a whole lot either...  I am up for suggestions.